src/standalone/run_s390x.go

```diff
@@ -0,0 +1,204 @@
+	config     *SecretConfig
+	dtclient   dtclient.Client
+	installer  installer.Installer
+	hostTenant string
+}
+
+func NewRunner(fs afero.Fs) (*Runner, error) {
+	log.Info("creating standalone runner")
+	config, err := newSecretConfigViaFs(fs)
+	if err != nil {
+		return nil, err
+	}
+	env, err := newEnv()
+	if err != nil {
+		return nil, err
+	}
+	client, err := newDTClientBuilder(config).createClient()
+	if err != nil {
+		return nil, err
+	}
+	oneAgentInstaller := url.NewUrlInstaller(
+		fs,
+		client,
+		&url.Properties{
+			Os:           dtclient.OsUnix,
+			Type:         dtclient.InstallerTypePaaS,
+			Flavor:       env.InstallerFlavor,
+			Arch:         env.InstallerArch,
+			Technologies: env.InstallerTech,
+			Version:      url.VersionLatest,
+			Url:          env.InstallerUrl,
+		},
+	)
+	log.Info("standalone runner created successfully")
+	return &Runner{
+		fs:        fs,
+		env:       env,
+		config:    config,
+		dtclient:  client,
+		installer: oneAgentInstaller,
+	}, nil
+}
+
+func (runner *Runner) Run() (resultedError error) {
+	log.Info("standalone agent init started")
+	defer runner.consumeErrorIfNecessary(&resultedError)
+
+	if err := runner.setHostTenant(); err != nil {
+		return err
+	}
+
+	if runner.env.Mode == InstallerMode && runner.env.OneAgentInjected {
+		if err := runner.installOneAgent(); err != nil {
+			return err
+		}
+		log.Info("OneAgent download finished")
+	}
+
+	err := runner.configureInstallation()
+	if err == nil {
+		log.Info("standalone agent init completed")
+	}
+	return err
+}
+
+func (runner *Runner) consumeErrorIfNecessary(resultedError *error) {
+	if !runner.env.CanFail && *resultedError != nil {
+		log.Error(*resultedError, "This error has been masked to not fail the container.")
+		*resultedError = nil
+	}
+}
+
+func (runner *Runner) setHostTenant() error {
+	log.Info("setting host tenant")
+	runner.hostTenant = NoHostTenant
+	if runner.config.HasHost {
+		hostTenant, ok := runner.config.MonitoringNodes[runner.env.K8NodeName]
+		if !ok {
+			return errors.Errorf("host tenant info is missing for %s", runner.env.K8NodeName)
+		}
+		runner.hostTenant = hostTenant
+	}
+	log.Info("successfully set host tenant", "hostTenant", runner.hostTenant)
+	return nil
+}
+
+// installOneAgent() must create the missing ruxitagentproc.conf on the s390x platform.
+// We must therefore inject the Agent ConnectionInfo into the general section of the config instance,
+// which will get merged into the config file after the first Update().
+// This is a temporary workaround until ruxitagentproc.conf gets restored to the zip installer for s390.
+func (runner *Runner) installOneAgent() error {
+	log.Info("downloading OneAgent")
+	err := runner.installer.InstallAgent(BinDirMount)
+	if err != nil {
+		return err
+	}
+	processModuleConfig, err := runner.dtclient.GetProcessModuleConfig(0)
+	if err != nil {
+		return err
+	}
+	connectionInfo, err := runner.dtclient.GetConnectionInfo()
+	if err != nil {
+		return err
+	}
```

Comment:
this is the only difference from the original, the difference is minimal, so just a simple `if arch.Arch == arch.ArchS390` should be enough

Reply:
@0sewa0 resolved this as requested.